src/components/NewCommentForm.tsx

```diff
@@ -1,6 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+import { Errors } from '../types/Errors';
+
+type Props = {
+  newCommentsData: CommentData;
+  setNewCommentsData: (data: CommentData) => void;
+  addComment: () => void;
+  isCommentAdding: boolean;
+  typeOfError: Errors;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  newCommentsData,
+  setNewCommentsData,
+  addComment,
+  isCommentAdding,
+  typeOfError,
+}) => {
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+
+  const inputCommentsName = (event: React.FormEvent<HTMLInputElement>) => {
+    const name = event.currentTarget.value;
+
+    setNewCommentsData({ ...newCommentsData, name });
+    setNameError(false);
+  };
+
+  const inputCommentsEmail = (event: React.FormEvent<HTMLInputElement>) => {
+    const email = event.currentTarget.value;
+
+    setNewCommentsData({ ...newCommentsData, email });
+    setEmailError(false);
+  };
+
+  const inputCommentsBody = (event: React.FormEvent<HTMLTextAreaElement>) => {
+    const body = event.currentTarget.value;
+
+    setNewCommentsData({ ...newCommentsData, body });
+    setBodyError(false);
+  };
+
+  const handleAddClick = () => (event: React.FormEvent<HTMLButtonElement>) => {
```

Comment:
handleAddClick should not be an arrow function returning another arrow function. Change it to a single arrow function.